Formula/w/wthrr.rb

```diff
@@ -0,0 +1,24 @@
+class Wthrr < Formula
+  desc "Weather Companion for the Terminal"
+  homepage "https://github.com/ttytm/wthrr-the-weathercrab"
+  url "https://github.com/ttytm/wthrr-the-weathercrab/archive/refs/tags/v1.2.0.tar.gz"
+  sha256 "6dbad5cf37fc364ec383f3c04cb4c676b0be8b760c06c0b689d1a04fc082c66c"
+  license "MIT"
+  head "https://github.com/ttytm/wthrr-the-weathercrab.git", branch: "main"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "openssl@3"
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system bin/"wthrr", "-h"
+    assert_match version.to_s, shell_output("#{bin}/wthrr --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I tried other subcommands, but error msg said it's not terminal...

This CLI returns weather info in a ASCII table. I'm not sure why this cause an error. 

I didn't find another subcommand other than -v or -h that returns no table.

Reply:
I will give it another try tomorrow. It seems `reset` might work.

Reply:
You might need to use `PTY.spawn` to simulate running inside a terminal. You can find many examples of doing this in this repository.